torch/distributed/fsdp/_wrap_utils.py

```diff
@@ -129,16 +129,24 @@ def _get_submodule_to_states(
     wrapped_modules.reverse()
     wrapped_modules_set = set(wrapped_modules)
     for submodule in wrapped_modules:
-        # Perform a BFS from `submodule` and record all unvisited state that is
-        # not already associated with another module in `wrapped_modules`.
-        queue: Deque[Tuple[nn.Module, str]] = collections.deque()
-        queue.append((submodule, ""))
+        # Perform a DFS from `submodule` and record all unvisited state that is
```

Comment:
I assume state dict save/load does not care about this, so changing the order here won't break BC?

Reply:
There are some subtle things happening with the traversal orders throughout this PR, which led to me having to rework some logic.

For this particular question, this should not affect state dict save and load. The only place where traversal order matters is which traversal order to use to have `state._handles` be ordered the same across wrapper and composable paths (which does not affect correctness but can improve the debugging experience). I had to rework some logic to achieve this. I will explain more details in the PR summary.